grpc-kotlin/src/test/kotlin/com/linecorp/armeria/server/grpc/kotlin/CoroutineServerInterceptorTest.kt

```diff
@@ -0,0 +1,317 @@
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @ValueSource(strings = ["/non-blocking", "/blocking"])
+    @ParameterizedTest
+    fun authorizedUnaryRequest(path: String) {
+        runTest {
+            val client = GrpcClients.builder(server.httpUri())
+                .auth(AuthToken.ofOAuth2(token))
+                .pathPrefix(path)
+                .build(TestServiceGrpcKt.TestServiceCoroutineStub::class.java)
+
+            assertThat(client.unaryCall(SimpleRequest.newBuilder().setFillUsername(true).build()).username)
+                .isEqualTo(username)
+        }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @ValueSource(strings = ["/non-blocking", "/blocking"])
+    @ParameterizedTest
+    fun unauthorizedUnaryRequest(path: String) {
+        runTest {
+            val client = GrpcClients.builder(server.httpUri())
+                .pathPrefix(path)
+                .build(TestServiceGrpcKt.TestServiceCoroutineStub::class.java)
+
+            runCatching {
+                client.unaryCall(SimpleRequest.newBuilder().setFillUsername(true).build())
+            }.onSuccess {
+                assert(false) { "Expected exception to be thrown, but none was thrown" }
+            }.onFailure { throwable ->
+                assert(throwable is StatusException)
+                val exception = throwable as StatusException
+                assert(exception.status == Status.UNAUTHENTICATED)
+            }
+        }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @ValueSource(strings = ["/non-blocking", "/blocking"])
+    @ParameterizedTest
+    fun authorizedStreamingOutputCall(path: String) {
+        runTest {
+            val client = GrpcClients.builder(server.httpUri())
+                .auth(AuthToken.ofOAuth2(token))
+                .pathPrefix(path)
+                .build(TestServiceGrpcKt.TestServiceCoroutineStub::class.java)
+
+            client.streamingOutputCall(StreamingOutputCallRequest.getDefaultInstance()).collect {
+                assertThat(it.payload.body.toStringUtf8()).isEqualTo(username)
+            }
+        }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @ValueSource(strings = ["/non-blocking", "/blocking"])
+    @ParameterizedTest
+    fun unauthorizedStreamingOutputCall(path: String) {
+        runTest {
+            val client = GrpcClients.builder(server.httpUri())
+                .pathPrefix(path)
+                .build(TestServiceGrpcKt.TestServiceCoroutineStub::class.java)
+            runCatching {
+                client.streamingOutputCall(StreamingOutputCallRequest.getDefaultInstance()).collect()
+            }.onSuccess {
+                assert(false) { "Expected exception to be thrown, but none was thrown" }
+            }.onFailure { throwable ->
+                assert(throwable is StatusException)
+                val exception = throwable as StatusException
+                assert(exception.status == Status.UNAUTHENTICATED)
```

Comment:
Should we use `assertThat` instead?

Reply:
Done in [b9f059c](https://github.com/line/armeria/pull/4724/commits/b9f059cc6a6e5efdb836aad7afb443882b658882)!